ir/type.cpp

```diff
@@ -608,7 +608,7 @@ StateValue AggregateType::aggregateVals(const vector<StateValue> &vals) const {
   StateValue v;
   for (unsigned idx = 0; idx < elements; ++idx) {
     auto vv = children[idx]->toBV(vals[idx]);
-    v = idx == 0 ? move(vv) : v.concat(vv);
```

Comment:
what do you think about making the representation dependent on the target's endianness. So, keep the old code for big endian, and add your code for little endian.
With this, we avoid byte swaps when loading/storing.

Reply:
The current format is wrong in the big endian too. In big endian, first element should come at the lower address as well. The `test/Transforms/ConstProp/loads.ll` test also guarantees this.